hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/events/FixedThreadPoolWithAffinityExecutor.java

```diff
@@ -154,15 +175,84 @@ public long scheduledEvents() {
     return scheduled.value();
   }
 
+  @Override
+  public long droppedEvents() {
+    return dropped.value();
+  }
+
   @Override
   public void close() {
+    isRunning.set(false);
     for (ThreadPoolExecutor executor : executors) {
       executor.shutdown();
     }
+    executorMap.clear();
+    DefaultMetricsSystem.instance().unregisterSource(EVENT_QUEUE + name);
   }
 
   @Override
   public String getName() {
     return name;
   }
+
+  /**
+   * Runnable class to perform execution of payload.
+   */
+  public static class ContainerReportProcessTask<P> implements Runnable {
+    private BlockingQueue<P> queue;
+    private AtomicBoolean isRunning;
+    private Map<String, FixedThreadPoolWithAffinityExecutor> executorMap;
+
+    public ContainerReportProcessTask(BlockingQueue<P> queue,
+        AtomicBoolean isRunning,
+        Map<String, FixedThreadPoolWithAffinityExecutor> executorMap) {
+      this.queue = queue;
+      this.isRunning = isRunning;
+      this.executorMap = executorMap;
+    }
+
+    @Override
+    public void run() {
+      while (isRunning.get()) {
+        try {
+          Object report = queue.poll(1, TimeUnit.MILLISECONDS);
```

Comment:
@sumitagrawl , what is the reason for setting such a short timeout?  It is like busy waiting; see #8431 .